.github/workflows/unit-test-mbed.yaml

```diff
@@ -0,0 +1,77 @@
+# Copyright (c) 2021 Project CHIP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: Build unit test - Mbed OS
+
+on:
+    push:
+    pull_request:
+
+concurrency:
+    group: ${{ github.ref }}-${{ github.workflow }}-${{ (github.event_name == 'pull_request' && github.event.number) || (github.event_name == 'workflow_dispatch' && github.run_number) || github.sha }}
+    cancel-in-progress: true    
+
+jobs:
+    mbedos-unittest:
+        strategy:
+            fail-fast: False
+            matrix:
+                APP_TARGET: [CY8CPROTO_062_4343W]
+        
+        name: "Unit tests: ${{matrix.APP_TARGET}}"
+        env:
+            BUILD_TYPE: mbedos
+            APP_PROFILE: release
+
+        runs-on: ubuntu-latest
+
+        container:
+            image: connectedhomeip/chip-build-mbed-os:latest
+            volumes:
+                - "/tmp/output_binaries:/tmp/output_binaries"
+
+        steps:
+            - name: Checkout
+              uses: actions/checkout@v2
+              # Fetch depth 0 to get all history and be able to check mergepoint for bloat report
+              with:
+                  fetch-depth: 0
+                  submodules: true
+
+            - name: Build unit test
+              run: scripts/tests/mbed/mbed_unit_tests.sh -c=build -b=${{matrix.APP_TARGET}} -p=$APP_PROFILE
```

Comment:
Could this be run as part of the `examples-mbed.yaml`? There's discussion on slack that we run out of computing power on the github runners and we should merge some jobs not to lose so much time on the checkout, bootstrap etc.

Reply:
I've seen that the time to run CI has gone up dramatically. I can look how feasible it is to integrate with the mbed os examples. In our development branch, `unit-test-mbed-yaml` is part of a more complex serie of workflow that cross compile chiptool and also run it on actual hardware.

Reply:
I think it is possible to integrate unit-test building with example-mbed workflow but first, we need to change it a little bit. Not using the build matrix but building examples as separate job steps. Then we can add the building unit-test as the next step.

Reply:
The PR that changes example-mbed workflow was created (#8902). After merge, we will be able to add unit-test building to it.

Reply:
PR was merged so I moved the unit-test building to examples-mbed workflow